lib/pharos/phases/validate_host.rb

```diff
@@ -74,6 +75,19 @@ def cpu_arch
           id: cpu['Architecture']
         )
       end
+
+      # @param interface [String]
+      # @return [String]
+      def private_interface_address(interface)
+        @ssh.exec!("ip -o addr show dev #{interface} scope global | awk '{ print $4 }'").each_line do |line|
```

Comment:
I would prefer to do output parsing completely in the code (easier to write specs).

Reply:
spec'd